libraries/AP_NMEA_Output/AP_NMEA_Output.cpp

```diff
@@ -101,7 +101,7 @@ void AP_NMEA_Output::update()
 
     // format latitude
     char lat_string[13];
-    float deg = fabsf(loc.lat * 1.0e-7f);
+    double deg = fabsF(loc.lat * 1.0e-7f);
```

Comment:
I think we should use fabs() and set #define ALLOW_DOUBLE_MATH_FUNCTIONS
otherwise this won't fix the issue on boards that don't enable double precision for the EKF, so it won't fix it on F4 boards

Reply:
In fact I reported back in discord, this PR isn't fixing the issue. When we multiply by 10e-7 we loose the 2 last decimal digits, they are rounded up, and there is where the issue comes from. Not even doing what you propose here fixes it. For some reason double can not hold that precision in the decimal values.

I was doing an approach of performing the logic to translate to NMEA without multiplying by 10e-7, and apparently it is working this way. I need to clean it up before pushing here though. What do you think? thanks!

Reply:
Okey, I just tested the approach of leaving it as it is now, adding the ALLOW_DOUBLE_MATH_FUNCTIONS ( with double and fabs() ) And it still rounds latest 2 decimal points.

I am force pushing the approach I tested to be working so far, working in deg 10e7 to not loose those decimals. Let me know your thoughts.